packages/modules/pv/set_values.py

```diff
@@ -0,0 +1,64 @@
+import os
+from pathlib import Path
+
+try:
+    from ...helpermodules import log
+    from ...helpermodules import pub
+except:
+    # for 1.9 compability
+    import sys
+    parentdir2 = str(Path(os.path.abspath(__file__)).parents[2])
+    sys.path.insert(0, parentdir2)
+    from helpermodules import log
+    from helpermodules import pub
+
+
+class set_values():
+    def __init__(self) -> None:
+        pass
+
+    def set(self, num, values, ramdisk):
+        """
+        Parameter
+        ---------
+        values: [power,
+                counter, (in Watt)
+                [current1, current2, current3]]
+        """
+        try:
+            if ramdisk == True:
+                self.write_to_ramdisk(values)
+            else:
+                self.pub_to_broker(num, values)
+        except Exception as e:
+            log.log_exception_comp(e, ramdisk)
+
+    def write_to_ramdisk(self, values):
+        try:
+            values[0] = [round(val, 1) for val in values[0]]
+            self.write_to_file("/pvwatt", int(values[0]))
```

Comment:
Hmm, es gibt ja die Möglichkeit, 2 PV Module zu konfigurieren. Wenn ich das hier richtig sehe, wäre das mit dem Modul nicht möglich, die Variablen für das 2. WR Modul zu füllen. Das ganze Verzeichnis für PV2 zu kopieren würde zu den  gleichen Problemen führen, die es in 1.9 auch schon gibt. Module, die nicht konsistent sind.

Reply:
Richtig, die WR2-Module fehlen noch. 
In der Set-Methode wird die Nummer des WR übergeben. Bei den Ladepunkten ist es bereits umgesetzt, dass die Nummer auch an `write_to_ramdisk ` übergeben wird, sodass dann der Name der Ramdisk-Datei entsprechend angepasst werden kann. 
```python
    def write_to_ramdisk(self, num, values):
        try:
            values[0] = [round(val, 1) for val in values[0]]
            values[1] = [round(val, 1) for val in values[1]]
            values[2] = round(values[2], 3)
            values[3] = int(values[3])
            if num <= 3:
                if num == 1:
                    suffix = ""
                elif num == 2:
                    suffix = "s1"
                    suffix2 = "lp2"
                elif num == 2:
                    suffix = "s2"
                    suffix2 = "lp3"
                self.write_to_file("/llv"+suffix+"1", values[0][0])
```
https://github.com/LKuemmel/openWB/blob/40be4688a2f026584a3f2317fc842ccd8bff31f8/packages/modules/cp/set_values.py#L40-L55

`write_to_ramdisk` wird von 1.9 genutzt, um die Daten in die Ramdisk zu schreiben; `pub_to_broker` von 2.0, um die Daten direkt an den Broker zu publishen.

Aktuell wird die Modulstruktur nochmal überarbeitet, dann folgen auch die WR2-Module!

Reply:
Dein Wunsch ist schon wahr, das drumherum ist noch nicht ganz fertig!